### ISSUE TITLE
[socket.io.js:2954]: Error: invalid payload

### ISSUE THREAD

**Issue by stratus-dev-fr** [socketio/socket.io#4392]
**Describe the bug**
When emitting the initialize function from the client, without params, it crashes with this fatal error sometimes.

**To Reproduce**

Reproduced randomly, we don't find how does it appear.

Please fill the following code example:

Socket.IO server version: `4.5.1`

*Server*

```js
private
```

Socket.IO client version: `4.5.1`

*Client*

```js
private
```

**Expected behavior**
The multiplayer does not work.

The entire error :
```
[socket.io.js:2887]: Error: invalid payload

decodeString@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:2887:19
add@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:2779:25
ondata@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:3982:22
Emitter.prototype.emit@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:583:22
onPacket@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:2226:20
Emitter.prototype.emit@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:583:22
onPacket@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:945:66
callback@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:1190:18
onData@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:1194:53
Emitter.prototype.emit@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:583:22
onData@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:1533:14
onLoad@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:1587:16
create/xhr.onreadystatechange@https://cdnjs.cloudflare.com/ajax/libs/socket.io/4.4.1/socket.io.js:1486:22
```

**Platform:**
I’m on macOS with Firefox 101.0.1 (on Brave macOS + on my iPhone with Chrome does not work too)
On iOS, even on Windows

Thanks a lot for your help and have a good day,
Corentin

**Comment by darrachequesne** [comment]
Hi!

After some digging, an `invalid payload` exception can be thrown [here](https://github.com/socketio/socket.io-parser/blob/c7514b5aa638d8b779c72a1172a2a694e97b15b7/lib/index.ts#L246) (called [here](https://github.com/socketio/socket.io-client/blob/f0350a0100f7261a31acb647f48261f4cad1f117/lib/manager.ts#L427)). Though this means that the server has sent an invalid packet. Which server implementation/version are you using?

**Comment by stratus-dev-fr** [comment]
Here is both errors that I have when it throw the `invalid payload` error : 

`GET wss://.../socket.io/?EIO=4&transport=websocket&sid=3Olimbg24Bj2VRpUAAGQ`
`Firefox can’t establish a connection to the server at wss://.../socket.io/?EIO=4&transport=websocket&sid=3Olimbg24Bj2VRpUAAGQ.`

Now, I have a friend with a different operator that have permanently the issue. Others does not have the issue 😬

**Comment by stratus-dev-fr** [comment]
Hi,

Some news about the issue : 

- The "invalid payload"'s issue disappeared permanently when we have reset our db (but why does my friend does not have the issue if there was the same db to everyone ?)
- But now, when I click on my input in order to upload a file on my server, it shows us the CORS's error "Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at xxx/socket.io/?EIO=4&transport=polling&t=NMnp2WI. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing).", same on https://socket.io/docs/v3/handling-cors/#Configuration

But, when I'm trying to do the command "curl "https://api.example.com/socket.io/?EIO=4&transport=polling"" on my computer with my config, I have the correct response, whereas when I try to upload my file on my website, it shows me the "Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at xxx/socket.io/?EIO=4&transport=polling&t=NMnp2WI. (Reason: CORS header ‘Access-Control-Allow-Origin’ missing)."'s error.

What can cause this issue ? Why does this error appear when I click on my input to upload whereas I only have an "onchange" event ?

Thanks a lot in advance,
Corentin

**Comment by Julien-Marcou** [comment]
Are you using a proxy (like Apache) on your server ?

This remind of a random "Firefox can’t establish a connection to the server" error I had.
The version of Apache I was using had an issue with sockets, updating Apache instantly solved my issue.

**Comment by darrachequesne** [comment]
This should be fixed by https://github.com/socketio/socket.io-client/commit/c5970231699aa47b00c4a617af4239d0fa90fa53, included in version [4.5.2](https://github.com/socketio/socket.io/releases/tag/4.5.2).

Thanks a lot for the detailed report :+1: 

Please reopen if needed.
